### PR TITLE
Configure eslint and prettier 

### PR DESCRIPTION
**Pull request by ibratoev**
Configure Prettier to format the code.
Configure ESLint to ignore all rules handled by Prettier.
Add an .editorconfig file as a common way to configure code editors. Prettier reads some of the editorconfig options (https://prettier.io/docs/en/configuration.html#editorconfig).
Configure the CI build to lint PRs.

Format all the code files using prettier.
Fix the single failing ESLint issue.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "arrowParens": "always",
+    "singleQuote": true,
+    "trailingComma": "all"
```
Comment (veerjd) [inline .prettierrc:4]: Is this the semicolon at the end? I prefer not to have it 😅

Comment (veerjd) [reply .prettierrc:4]: I checked and it's a matter of adding "semi": false here, if you are ok with that

**Review by veerjd — COMMENTED** [review]
Can we remove semicolons at the end of lines in the linters?

File: .prettierrc (hunk 1 above)
Comment (ibratoev) [reply .prettierrc:4]: I prefer having semicolons but I am fine with not having. Up to you here ;)
Prettier reformats everything so it is easy to use any way.

**Review by veerjd — APPROVED** [review]

**Pull request merged**